### PR TITLE
daily job running mobile example on real devices

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Test mobile example on real devices

## Solution

- Use [BrowserStack](https://www.browserstack.com) to have access to [real devices](https://www.browserstack.com/list-of-browsers-and-platforms/app_automate)
  - [App Automate](https://www.browserstack.com/app-automate) to run the example
  - [App Percy](https://www.browserstack.com/app-percy) to compare the screenshot
- Added a daily/manual CI job that will build for iOS and Android, send the apps to BrowserStack, run the app on one iOS device and one Android device, capture a screenshot, send it for visual validation, and archive it in the GitHub action

Example run: https://github.com/mockersf/bevy/actions/runs/4521883534

They currently have a bug with the settings to view snapshots, they should be public. I'll raise it to them, and if they don't fix it in time it's possible to work around for everyone to view the results through their API.

@cart to get this to work, you'll need
- to set up an account on BrowserStack
- add the secrets `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` to the Bevy repo
- create a project in Percy
- add the secret `PERCY_TOKEN` to the Bevy repo and modify the project name line 122 in the `Daily.yml` file

### CODE REVIEW

File: examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -229,6 +229,7 @@
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				CODE_SIGN_IDENTITY = "";
```
Comment (mockersf) [inline examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj:232]: `CODE_SIGN_IDENTITY = ""` is needed to be able to build the app without signing it in CI

File: examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -293,7 +294,7 @@
 					"-lbevy_mobile_example",
 					"-lc++abi",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = "com.rust.bevy_mobile_example";
+				PRODUCT_BUNDLE_IDENTIFIER = "org.bevyengine.example";
```
Comment (mockersf) [inline examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj:297]: this is so that both Android and iOS have the same package name

**Comment by alice-i-cecile** [comment]
Controversial label added because of the third party integration. Strongly in favor of this though: there's a been quite a few surprising regressions, and if we're going to get mobile support working in earnest we need to be testing on real devices.

File: examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj (hunk 1 above)
Comment (alice-i-cecile) [reply examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj:232]: Can you leave a comment in this type of file?

Comment (mockersf) [reply examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj:232]: The file is not meant to be read, it's generated and managed through xcode. It's also a setting specific to Bevy as an engine, someone creating a game and targeting iOS will have a signing identity. As Bevy is not paying the 100$/year apple development subscription, we don't have one.

**Review by cart — APPROVED** [review]

**Comment by cart** [comment]
(@mockersf sent me the secrets. I added them to the Bevy Org and gave the bevy repo access)

File: README.md [hunk 3]
```diff
@@ -107,6 +107,8 @@ Plugins are very welcome to extend Bevy's features. [Guidelines][plugin_guidelin
 
 Additionally, we would like to thank the [Amethyst](https://github.com/amethyst/amethyst), [macroquad](https://github.com/not-fl3/macroquad), [coffee](https://github.com/hecrj/coffee), [ggez](https://github.com/ggez/ggez), [Fyrox](https://github.com/FyroxEngine/Fyrox), and [Piston](https://github.com/PistonDevelopers/piston) projects for providing solid examples of game engine development in Rust. If you are looking for a Rust game engine, it is worth considering all of your options. Each engine has different design goals, and some will likely resonate with you more than others.
 
+This project is tested with BrowserStack.
```
Comment (alice-i-cecile) [inline README.md:110]:
```suggestion
This project is tested on mobile devices with [BrowserStack](https://www.browserstack.com/).
```

I think this is genuinely clearer and more helpful to readers.

**Review by alice-i-cecile — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (mockersf) [reply README.md:110]: They asked for the exact string to be present in the readme. My guess is that they have a tool checking for its presence

Comment (mockersf) [reply README.md:111]: added a comment to make it clearer

**Pull request merged**